local/local_test.go

```diff
@@ -540,42 +491,71 @@ func testLocalImage(t *testing.T, when spec.G, it spec.S) {
-						FROM busybox
-						LABEL repo_name_for_randomisation=%s
-						RUN echo new-base > base.txt
-						RUN echo text-new-base > otherfile.txt
-					`, newBase), nil)
-				}()
+				// new base image
+				newBase = "pack-newbase-test-" + h.RandString(10)
+
+				newBaseImage, err := local.NewImage(newBase, dockerClient, local.FromBaseImage("busybox"))
+
+				h.AssertNil(t, newBaseImage.SetLabel("repo_name_for_randomisation", repoName))
+
+				newBaseLayer1Path, err := h.CreateSingleFileTar("/base.txt", "new-base")
+				h.AssertNil(t, err)
+				defer os.Remove(newBaseLayer1Path)
 
+				newBaseLayer2Path, err := h.CreateSingleFileTar("/otherfile.txt", "text-new-base")
+				h.AssertNil(t, err)
+				defer os.Remove(newBaseLayer2Path)
+
+				h.AssertNil(t, newBaseImage.AddLayer(newBaseLayer1Path))
+				h.AssertNil(t, newBaseImage.AddLayer(newBaseLayer2Path))
+
+				h.AssertNil(t, newBaseImage.Save())
+
+				// old base image
 				oldBase = "pack-oldbase-test-" + h.RandString(10)
-				h.CreateImageOnLocal(t, dockerClient, oldBase, fmt.Sprintf(`
-					FROM busybox
-					LABEL repo_name_for_randomisation=%s
-					RUN echo old-base > base.txt
-					RUN echo text-old-base > otherfile.txt
-				`, oldBase), nil)
+				oldBaseImage, err := local.NewImage(oldBase, dockerClient, local.FromBaseImage("busybox"))
+
+				h.AssertNil(t, oldBaseImage.SetLabel("repo_name_for_randomisation", repoName))
+
+				oldBaseLayer1Path, err := h.CreateSingleFileTar("/base.txt", "old-base")
+				h.AssertNil(t, err)
+				defer os.Remove(oldBaseLayer1Path)
+
+				oldBaseLayer2Path, err := h.CreateSingleFileTar("/otherfile.txt", "text-old-base")
+				h.AssertNil(t, err)
+				defer os.Remove(oldBaseLayer2Path)
+
+				h.AssertNil(t, oldBaseImage.AddLayer(oldBaseLayer1Path))
+				h.AssertNil(t, oldBaseImage.AddLayer(oldBaseLayer2Path))
+
+				h.AssertNil(t, oldBaseImage.Save())
+
 				inspect, _, err := dockerClient.ImageInspectWithRaw(context.TODO(), oldBase)
 				h.AssertNil(t, err)
 				oldTopLayer = inspect.RootFS.Layers[len(inspect.RootFS.Layers)-1]
 
-				h.CreateImageOnLocal(t, dockerClient, repoName, fmt.Sprintf(`
-					FROM %s
-					LABEL repo_name_for_randomisation=%s
-					RUN echo text-from-image > myimage.txt
-					RUN echo text-from-image > myimage2.txt
-				`, oldBase, repoName), nil)
+				// original image
+				origImage, err := local.NewImage(repoName, dockerClient, local.FromBaseImage(oldBase))
+
+				h.AssertNil(t, origImage.SetLabel("repo_name_for_randomisation", repoName))
+
+				imgLayer1Path, err := h.CreateSingleFileTar("/myimage.txt", "text-from-image")
+				h.AssertNil(t, err)
+				defer os.Remove(imgLayer1Path)
+
+				imgLayer2Path, err := h.CreateSingleFileTar("/myimage2.txt", "text-from-image")
+				h.AssertNil(t, err)
+				defer os.Remove(imgLayer2Path)
+
+				h.AssertNil(t, origImage.AddLayer(imgLayer1Path))
+				h.AssertNil(t, origImage.AddLayer(imgLayer2Path))
+
+				h.AssertNil(t, origImage.Save())
+
```

Comment:
There are more lines here. Should we reduce this test case to 1 layer per image?